test/torchaudio_unittest/transforms/autograd_test_impl.py

```diff
@@ -125,6 +125,30 @@ def test_fade(self, fade_shape):
         waveform = get_whitenoise(sample_rate=8000, duration=0.05, n_channels=2)
         self.assert_grad(transform, [waveform], nondet_tol=1e-10)
 
+    @parameterized.expand([(T.TimeMasking,), (T.FrequencyMasking,)])
+    def test_masking(self, masking_transform):
+        sample_rate = 8000
+        n_fft = 400
+        spectrogram = get_spectrogram(
+            get_whitenoise(sample_rate=sample_rate, duration=0.05, n_channels=2),
+            n_fft=n_fft, power=1)
+        deterministic_transform = _DeterministicWrapper(masking_transform(400))
+        self.assert_grad(deterministic_transform, [spectrogram])
+
+    @parameterized.expand([(T.TimeMasking,), (T.FrequencyMasking,)])
+    def test_masking_iid(self, masking_transform):
+        sample_rate = 8000
+        n_fft = 400
+        specs = [get_spectrogram(
+            get_whitenoise(sample_rate=sample_rate, duration=0.05, n_channels=2, seed=i),
+            n_fft=n_fft, power=1)
+            for i in range(3)
+        ]
+
+        batch = torch.cat(specs)
```

Comment:
Can you add assertion in the `assert batch.ndim == 4`?
The code in the question automatically falls back to non-iid code path, so having assertion here makes it simpler to review the code.

https://github.com/pytorch/audio/blob/1f136671b84071a2fe1d5b762df64f3a76310c31/torchaudio/transforms.py#L884

Reply:
Done!